DataBiosphere/leonardo
# IA-2994 use built-in stopCluster

https://broadworkbench.atlassian.net/browse/IA-2994

Related PR https://github.com/broadinstitute/workbench-libs/pull/783

Dataproc started supporting "stop" a cluster a while ago. This means we can use sdk's stopCluster API directly to stop a cluster, which should save on dataproc premium when users intend to stop a cluster, because today, when user stops a cluster, we manually stops all instances of the cluster, but the cluster itself is still in `RUNNING` status.

However, during a patch update, it is still useful to not have a full cluster stop, hence I introduced a `isDataprocFullStop` flag to indicate whether we intend to also stop the cluster in addition to shutting down all VM instances.

Also made some optimizations in `DataprocMonitor` so that we're not querying instance's status all the time when not needed (for instance, when we create a cluster, if the cluster is still in `Creating`, we don't need to poll individual VMs at that point, but we poll all instances to get VM status regardless of cluster status today)

Separately upgrade http4s to 0.23.6, which has a retry bug fix.

Created https://broadworkbench.atlassian.net/browse/IA-2999 to reflect the cost reduction in UI

Manual tests:
* Create a dataproc cluster via local terra UI. Create/stop/start works as expected
* Add pre-emptibles, and start/stop cluster works as expected

---
Have you read [CONTRIBUTING.md](https://github.com/DataBiosphere/leonardo/blob/develop/CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:


In all cases:
